src/config/watcher.rs

```diff
@@ -19,11 +22,49 @@ const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1
 
 const RETRY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
 
+/// Refer to [`crate::cli::WatchConfigMethod`] for details.
+pub enum WatcherConfig {
+    /// Recommended watcher for the current OS.
+    RecommendedWatcher,
+    /// A poll-based watcher that checks for file changes at regular intervals.
+    PollWatcher(u64),
+}
+
+enum Watcher {
+    /// recommended watcher for os, usually inotify for linux based systems
+    RecommendedWatcher(notify::RecommendedWatcher),
+    /// poll based watcher. for watching files from NFS.
+    PollWatcher(notify::PollWatcher),
+}
+
+impl Watcher {
+    fn add_paths(&mut self, config_paths: &[PathBuf]) -> Result<(), Error> {
+        for path in config_paths {
+            self.watch(path, RecursiveMode::Recursive)?;
+        }
+        Ok(())
+    }
+
+    fn watch(&mut self, path: &Path, recursive_mode: RecursiveMode) -> Result<(), Error> {
+        use notify::Watcher as NotifyWatcher;
```

Comment:
nit: any reason to use this here vs at the top like the other imports?

Reply:
Actually we already have name watcher as enum. it was colliding with watcher from `notify`.
Renaming the import at the top level user might not able to understand what is `NotifyWathcher` and we need the methods for `notify::Watcher` on that specific scope. that's why i used Watcher on the specific scope